packages/git/src/browser/diff/git-diff-widget.ts

```diff
@@ -35,6 +39,17 @@ export class GitDiffWidget extends GitBaseWidget<GitFileChangeNode> implements S
 
         this.addClass('theia-git');
 
+        this.toDispose.push(this.gitWatcher.onGitEvent(async gitEvent => {
```

Comment:
What is the requirement, should it track the repository changes too? What would happen if I open the diff widget for repository A, and then I switch to repository B?

Reply:
Good question. Same for the GitHistoryView. Should we close related diff and history views if we change the repo?

Reply:
We decided to keep the views after switching the repo.

Reply:
Thanks for the update on this. Out of curiosity and just to make sure I understood the requirements correctly; if I have two repositories (`A` and `B`), I open a diff for something in repo `A` and if I have any changes in repo `A`, I update the view content. Sounds good so far. Then I switch to repository `B`, but we keep the view. When I have any changes in repository `B`, I need to update the view, that's the contract, correct? So I call `setContent` but the repository provider has `B` as the active repository (and not `A` anymore) while the `options` is still something for repository `A`, right? Is that you want to support, or I did miss a step?

Reply:
Yes, that is what I want to support.

Reply:
Please do it in the separate PR.